docs/book/src/multiversion-tutorial/testdata/project/api/v2/disk_types.go

```diff
@@ -0,0 +1,130 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	diskapiv1 "infra.kubebuilder.io/api/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+/*
+We realized that the price field can be represented in a better way, so in
+next iteration v2 of our disk API, we changed the price to a structured field.
+*/
+
+// DiskSpec defines the desired state of Disk
+type DiskSpec struct {
+	// Price represents price per GB for the Disk.
+	Price Price `json:"price"`
+}
+
+// Price represents a generic price value that has amount and currency.
+type Price struct {
+	Amount   int64  `json:"amount"`
+	Currency string `json:"currency"`
+}
+
+// DiskStatus defines the observed state of Disk
```

Comment:
We can collapse from `DiskStatus` to `type Disk struct` to focus on what really matter.

Reply:
I tried doing it, but it isn't working as expected. Let's chat offline to debug it.